packages/theming/apple-theme/macos/AppleThemeModule.swift

```diff
@@ -0,0 +1,155 @@
+import Foundation
+import FluentUI
+
+@objc(MSFAppleThemeModule)
+class AppleThemeModule: NSObject {
+
+	static func themingModule() -> [AnyHashable : Any] {
+		return [
+			"typography" : [
+				"sizes" : fontSizes(),
+				"weights" : fontWeights(),
+				"families" : fontFamilies(),
+				"variants" : fontVariants()
+			],
+		]
+	}
+
+	@objc(getApplePartialThemeWithCallback:)
+	func getApplePartialTheme(callback: RCTResponseSenderBlock) {
+		callback([NSNull(), AppleThemeModule.themingModule()])
+	}
+	
+	// MARK: - Typography
+
+	static func fontFamilies() -> [AnyHashable : Any] {
+		if #available(OSX 10.15, *) {
+			return [
+				"primary" : NSFont.systemFont(ofSize: 0).familyName ?? "System",
+				"monospace" : NSFont.monospacedSystemFont(ofSize: 0, weight: .regular).familyName ?? "System",
+			]
+		} else {
+			return [
+				"primary" : NSFont.systemFont(ofSize: 0).familyName ?? "System",
+			]
+		}
+	}
+
+	/// Map the current FluentUI React Native font sizes approximately to their corresponding apple text style,
+	/// For older vesrions of macOS, fallback to the values described in the apple HIG:
+	/// https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/typography/
+	/// These mappings and variants are subject to change as we moved to a unified cross platform Fluent typography ramp
+	static func fontSizes() -> [AnyHashable : Any] {
+		if #available(OSX 11.0, *) {
+			return [
+				"caption" : NSFont.preferredFont(forTextStyle: .caption1, options: [:]).pointSize,
```

Comment:
are these weights all regular?

Reply:
No idea what the weights returns since it's the system API. However, in the method "fontVariants()" you can see me basically define a bunch of text styles with weights
```
			"captionStandard": [
				"face" : families["primary"],
				"size" : sizes["caption"],
				"weight" : weights[.medium],
			],
			"secondaryStandard": [
				"face" : families["primary"],
				"size" : sizes["secondary"],
				"weight" : weights[.regular],
			],
			///
```